--- a/truffle/mx.truffle/suite.py
+++ b/truffle/mx.truffle/suite.py
@@ -293,7 +293,9 @@
         "sdk:POLYGLOT",
         "sdk:NATIVEBRIDGE",
         "com.oracle.truffle.api.bytecode",
-        "com.oracle.truffle.api.impl.asm",
+        "truffle:ASM_9.8",
+        "truffle:ASM_TREE_9.8",
+        "truffle:ASM_COMMONS_9.8"
       ],
       "requires" : [
         "java.logging",
@@ -377,7 +379,9 @@
       "sourceDirs" : ["src"],
       "dependencies" : [
         "com.oracle.truffle.api.exception",
-        "com.oracle.truffle.api.impl.asm",
+        "truffle:ASM_9.8",
+        "truffle:ASM_TREE_9.8",
+        "truffle:ASM_COMMONS_9.8"
       ],
       "requires" : [
         "java.sql",
@@ -779,7 +783,9 @@
       "sourceDirs" : ["src"],
       "dependencies" : [
         "com.oracle.truffle.api",
-        "com.oracle.truffle.api.impl.asm",
+        "truffle:ASM_9.8",
+        "truffle:ASM_TREE_9.8",
+        "truffle:ASM_COMMONS_9.8"
       ],
       "requires" : [
         "jdk.unsupported", # sun.misc.Unsafe
@@ -1720,34 +1726,6 @@
       "graalCompilerSourceEdition": "ignore",
     },
 
-    "com.oracle.truffle.api.impl.asm" : {
-      # Shadowed ASM libraries (org.ow2.asm:asm,asm-tree,asm-commons)
-      "subDir" : "src",
-      "sourceDirs" : ["src"],
-      "javaCompliance" : "17+",
-      "spotbugsIgnoresGenerated" : True,
-      "shadedDependencies" : [
-        "truffle:ASM_9.8",
-        "truffle:ASM_TREE_9.8",
-        "truffle:ASM_COMMONS_9.8",
-      ],
-      "class" : "ShadedLibraryProject",
-      "shade" : {
-        "packages" : {
-          "org.objectweb.asm" : "com.oracle.truffle.api.impl.asm",
-        },
-        "exclude" : [
-          "META-INF/MANIFEST.MF",
-          "**/package.html",
-        ],
-      },
-      "description" : "ASM library shadowed for Truffle.",
-      # We need to force javac because the generated sources in this project produce warnings in JDT.
-      "forceJavac" : "true",
-      "javac.lint.overrides" : "none",
-      "jacoco" : "exclude",
-      "graalCompilerSourceEdition": "ignore",
-    },
 
     "org.graalvm.shadowed.org.jcodings" : {
       # Shadowed JCODINGS library (org.jruby.jcodings:jcodings)
@@ -2052,19 +2030,23 @@
     },
 
     "TRUFFLE_API" : {
+      "exclude": ["ASM_9.8", "ASM_TREE_9.8", "ASM_COMMONS_9.8"],
       # This distribution defines a module.
       "moduleInfo" : {
         "name" : "org.graalvm.truffle",
         "requires" : [
           "static java.desktop",
-          "jdk.unsupported", # sun.misc.Unsafe
+          "jdk.unsupported",
           "java.logging",
           "java.management",
           "jdk.management",
-          "java.sql", # java.sql.date java.sql.Time
+          "java.sql",
           "org.graalvm.collections",
           "org.graalvm.nativeimage",
           "static org.graalvm.nativebridge",
+          "org.objectweb.asm",
+          "org.objectweb.asm.tree",
+          "org.objectweb.asm.commons"
         ],
         "exports" : [
           # Unqualified exports
@@ -2416,6 +2398,7 @@
     },
 
     "TRUFFLE_DSL_PROCESSOR" : {
+      "exclude": ["ANTLR4"],
       "subDir" : "src",
       "dependencies" : [
         "com.oracle.truffle.dsl.processor",
--- a/truffle/src/com.oracle.truffle.api.bytecode.test/src/com/oracle/truffle/api/bytecode/test/InstructionBytecodeSizeTest.java
+++ b/truffle/src/com.oracle.truffle.api.bytecode.test/src/com/oracle/truffle/api/bytecode/test/InstructionBytecodeSizeTest.java
@@ -543,7 +543,7 @@
         public static int getMethodCodeSize(Method method) {
             Class<?> owner = method.getDeclaringClass();
             String name = method.getName();
-            String descriptor = com.oracle.truffle.api.impl.asm.Type.getMethodDescriptor(method);
+            String descriptor = org.objectweb.asm.Type.getMethodDescriptor(method);
             return getMethodCodeSize(owner, name, descriptor);
         }
 
--- a/truffle/src/com.oracle.truffle.api.staticobject.test/src/com/oracle/truffle/api/staticobject/test/HiddenClassGenerator.java
+++ b/truffle/src/com.oracle.truffle.api.staticobject.test/src/com/oracle/truffle/api/staticobject/test/HiddenClassGenerator.java
@@ -40,18 +40,18 @@
  */
 package com.oracle.truffle.api.staticobject.test;
 
-import com.oracle.truffle.api.impl.asm.ClassWriter;
-import com.oracle.truffle.api.impl.asm.MethodVisitor;
-import com.oracle.truffle.api.impl.asm.Type;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PUBLIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SUPER;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ALOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.BIPUSH;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INVOKESPECIAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IRETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.RETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.V1_6;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.BIPUSH;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.IRETURN;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.V1_6;
 
 public class HiddenClassGenerator {
     public static byte[] getBytes() {
--- a/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/ArrayBasedShapeGenerator.java
+++ b/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/ArrayBasedShapeGenerator.java
@@ -52,54 +52,54 @@
 import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
 import com.oracle.truffle.api.TruffleLanguage;
 import com.oracle.truffle.api.TruffleOptions;
-import com.oracle.truffle.api.impl.asm.AnnotationVisitor;
-import com.oracle.truffle.api.impl.asm.ClassVisitor;
-import com.oracle.truffle.api.impl.asm.ClassWriter;
-import com.oracle.truffle.api.impl.asm.Label;
-import com.oracle.truffle.api.impl.asm.MethodVisitor;
-import com.oracle.truffle.api.impl.asm.Opcodes;
-import com.oracle.truffle.api.impl.asm.Type;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
 import org.graalvm.collections.Pair;
 import org.graalvm.nativeimage.ImageInfo;
 
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_FINAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PRIVATE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PUBLIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_STATIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SUPER;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SYNTHETIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACONST_NULL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ALOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ANEWARRAY;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ARETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ASTORE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ATHROW;
-import static com.oracle.truffle.api.impl.asm.Opcodes.CHECKCAST;
-import static com.oracle.truffle.api.impl.asm.Opcodes.DOUBLE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.DUP;
-import static com.oracle.truffle.api.impl.asm.Opcodes.FLOAT;
-import static com.oracle.truffle.api.impl.asm.Opcodes.F_FULL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.GETFIELD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.GETSTATIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.GOTO;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IFEQ;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IFLE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IFNE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IFNONNULL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.IFNULL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ILOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INTEGER;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INVOKESPECIAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INVOKESTATIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INVOKEVIRTUAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.LONG;
-import static com.oracle.truffle.api.impl.asm.Opcodes.NEW;
-import static com.oracle.truffle.api.impl.asm.Opcodes.NEWARRAY;
-import static com.oracle.truffle.api.impl.asm.Opcodes.POP;
-import static com.oracle.truffle.api.impl.asm.Opcodes.PUTFIELD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.RETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.T_BYTE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.V1_8;
+import static org.objectweb.asm.Opcodes.ACC_FINAL;
+import static org.objectweb.asm.Opcodes.ACC_PRIVATE;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ACC_SYNTHETIC;
+import static org.objectweb.asm.Opcodes.ACONST_NULL;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ANEWARRAY;
+import static org.objectweb.asm.Opcodes.ARETURN;
+import static org.objectweb.asm.Opcodes.ASTORE;
+import static org.objectweb.asm.Opcodes.ATHROW;
+import static org.objectweb.asm.Opcodes.CHECKCAST;
+import static org.objectweb.asm.Opcodes.DOUBLE;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.FLOAT;
+import static org.objectweb.asm.Opcodes.F_FULL;
+import static org.objectweb.asm.Opcodes.GETFIELD;
+import static org.objectweb.asm.Opcodes.GETSTATIC;
+import static org.objectweb.asm.Opcodes.GOTO;
+import static org.objectweb.asm.Opcodes.IFEQ;
+import static org.objectweb.asm.Opcodes.IFLE;
+import static org.objectweb.asm.Opcodes.IFNE;
+import static org.objectweb.asm.Opcodes.IFNONNULL;
+import static org.objectweb.asm.Opcodes.IFNULL;
+import static org.objectweb.asm.Opcodes.ILOAD;
+import static org.objectweb.asm.Opcodes.INTEGER;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.INVOKEVIRTUAL;
+import static org.objectweb.asm.Opcodes.LONG;
+import static org.objectweb.asm.Opcodes.NEW;
+import static org.objectweb.asm.Opcodes.NEWARRAY;
+import static org.objectweb.asm.Opcodes.POP;
+import static org.objectweb.asm.Opcodes.PUTFIELD;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.T_BYTE;
+import static org.objectweb.asm.Opcodes.V1_8;
 
 final class ArrayBasedShapeGenerator<T> extends ShapeGenerator<T> {
     private static final ConcurrentHashMap<Pair<Class<?>, Class<?>>, Object> generatorCache = TruffleOptions.AOT ? new ConcurrentHashMap<>() : null;
--- a/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/FieldBasedShapeGenerator.java
+++ b/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/FieldBasedShapeGenerator.java
@@ -40,28 +40,28 @@
  */
 package com.oracle.truffle.api.staticobject;
 
-import com.oracle.truffle.api.impl.asm.ClassVisitor;
-import com.oracle.truffle.api.impl.asm.ClassWriter;
-import com.oracle.truffle.api.impl.asm.MethodVisitor;
-import com.oracle.truffle.api.impl.asm.Type;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.Map.Entry;
 
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_FINAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PUBLIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SUPER;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SYNTHETIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ALOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ARETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.DUP;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ILOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.INVOKESPECIAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.NEW;
-import static com.oracle.truffle.api.impl.asm.Opcodes.RETURN;
-import static com.oracle.truffle.api.impl.asm.Opcodes.V1_8;
+import static org.objectweb.asm.Opcodes.ACC_FINAL;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ACC_SYNTHETIC;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ARETURN;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.ILOAD;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.NEW;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.V1_8;
 
 final class FieldBasedShapeGenerator<T> extends ShapeGenerator<T> {
     private final GeneratorClassLoaders gcls;
--- a/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/ShapeGenerator.java
+++ b/truffle/src/com.oracle.truffle.api.staticobject/src/com/oracle/truffle/api/staticobject/ShapeGenerator.java
@@ -41,9 +41,9 @@
 package com.oracle.truffle.api.staticobject;
 
 import com.oracle.truffle.api.TruffleLanguage;
-import com.oracle.truffle.api.impl.asm.ClassVisitor;
-import com.oracle.truffle.api.impl.asm.FieldVisitor;
-import com.oracle.truffle.api.impl.asm.Type;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.Type;
 import com.oracle.truffle.api.staticobject.StaticShape.StorageStrategy;
 
 import java.lang.reflect.Field;
@@ -51,8 +51,8 @@
 import java.util.Map.Entry;
 import sun.misc.Unsafe;
 
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_FINAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_FINAL;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
 
 abstract class ShapeGenerator<T> {
     protected static final Unsafe UNSAFE = getUnsafe();
--- a/truffle/src/com.oracle.truffle.host/src/com/oracle/truffle/host/HostAdapterBytecodeGenerator.java
+++ b/truffle/src/com.oracle.truffle.host/src/com/oracle/truffle/host/HostAdapterBytecodeGenerator.java
@@ -40,16 +40,16 @@
  */
 package com.oracle.truffle.host;
 
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_FINAL;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PRIVATE;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_PUBLIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_STATIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_SUPER;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ACC_VARARGS;
-import static com.oracle.truffle.api.impl.asm.Opcodes.ALOAD;
-import static com.oracle.truffle.api.impl.asm.Opcodes.H_INVOKESTATIC;
-import static com.oracle.truffle.api.impl.asm.Opcodes.RETURN;
-import static com.oracle.truffle.api.impl.asm.Type.BOOLEAN_TYPE;
+import static org.objectweb.asm.Opcodes.ACC_FINAL;
+import static org.objectweb.asm.Opcodes.ACC_PRIVATE;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ACC_VARARGS;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.H_INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Type.BOOLEAN_TYPE;
 
 import java.lang.annotation.Annotation;
 import java.lang.invoke.CallSite;
@@ -67,13 +67,13 @@
 
 import org.graalvm.polyglot.Value;
 
-import com.oracle.truffle.api.impl.asm.ClassWriter;
-import com.oracle.truffle.api.impl.asm.FieldVisitor;
-import com.oracle.truffle.api.impl.asm.Handle;
-import com.oracle.truffle.api.impl.asm.Label;
-import com.oracle.truffle.api.impl.asm.Opcodes;
-import com.oracle.truffle.api.impl.asm.Type;
-import com.oracle.truffle.api.impl.asm.commons.InstructionAdapter;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.InstructionAdapter;
 
 /**
  * Generates bytecode for a Java adapter class. Used by the {@link HostAdapterFactory}.
--- a/truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/JDKSupport.java
+++ b/truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/JDKSupport.java
@@ -78,10 +78,10 @@
 import com.oracle.truffle.api.impl.Accessor;
 import com.oracle.truffle.api.impl.Accessor.JavaLangSupport;
 import com.oracle.truffle.api.impl.Accessor.ModulesAccessor;
-import com.oracle.truffle.api.impl.asm.ClassWriter;
-import com.oracle.truffle.api.impl.asm.FieldVisitor;
-import com.oracle.truffle.api.impl.asm.MethodVisitor;
-import com.oracle.truffle.api.impl.asm.Opcodes;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
 import jdk.internal.access.JavaLangAccess;
 import jdk.internal.access.SharedSecrets;
 import jdk.internal.module.Modules;
